### PR TITLE
Pull request workflow

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

File: adr/0001-pull-request-workflow.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Date: 2017-10-20
+
+## Status
+
+Accepted
+
+## Context
+
+Software is build incrementally as an accumulation of changes.
```
Comment (seanwiseman) [inline adr/0001-pull-request-workflow.md]: build -> built

**Review by seanwiseman — APPROVED** [review]

File: adr/0001-pull-request-workflow.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+We will open pull requests as the primary mean for deploying a change.
+
+## Consequences
+
+Team members can be asked to review a pull request through Github.
+
+Automated tests should run on pull requests, to provide automated feedback.
+
+Committing directly to a mainline branch (`develop` or `master`) should not be frequent.
```
Comment (thewilkybarkid) [inline adr/0001-pull-request-workflow.md]: '... only be done when necessary (eg infrastructure changes)'?

Comment (giorgiosironi) [reply adr/0001-pull-request-workflow.md]: Also formulas now have pull request workflows, but there are some that can't be built like builder-private. Including the explanation.

**Review by BlueReZZ — COMMENTED** [review]
While I understand what you're saying about protecting the mainline branch through pull requests, the wording suggests that this might not happen frequently, which seems counter intuitive to continuous delivery. Perhaps wording suggesting it should be less frequent relative to commits to PRs.

**Review by BlueReZZ — APPROVED** [review]

**Review by thewilkybarkid — APPROVED** [review]

**Review by gnott — APPROVED** [review]
Reads clear and simple

**Comment by lsh-0** [comment]
I'm not a fan of these documents ~~, I feel they generate temporary enthusiasm that real change is happening, cooperation is being achieved and that this document will be different to all of the other documents that have accumulated before it and, unlike all those documents, this one won't be forgotten about. This one isn't wasting our time and this one is definitely laying the foundation for a glorious library of write-once, read-never documentation.~~

> ... we should be wary of feature branches remaining open for too long without being merged into the mainline

we are all aware this happens. if this is an actual problem, enshrine the solution in a process: have alfred update a PR request after a reasonable period with a "this PR is too damn old @insertusernamehere !" and then close it after another reasonable period.

just as old PRs could be detected, have something that detects commits directly to the master/develop branches and emails an admonition to the one who did it, pointing them to this document.

**Comment by giorgiosironi** [comment]
Note the intention is not to prescribe, but to document what already happens, for example for onboarding or sharing outside the organization. Currently this information about design decisions is dumped in Jira or Google docs, which are unwieldly to update:

> Large documents are never kept up to date. Small, modular documents have at least a chance at being updated.

**Pull request merged**